daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java

```diff
@@ -488,24 +505,18 @@ protected void configure() {
             container.discoverComponents(extension.getClassRealm(), new SessionScopeModule(container),
                     new MojoExecutionScopeModule(container));
         }
-
-        eventSpyDispatcher = container.lookup(EventSpyDispatcher.class);
-
-        maven = container.lookup(Maven.class);
-
-        executionRequestPopulator = container.lookup(MavenExecutionRequestPopulator.class);
-
-        modelProcessor = createModelProcessor(container);
-
-        configurationProcessors = container.lookupMap(ConfigurationProcessor.class);
-
-        toolchainsBuilder = container.lookup(ToolchainsBuilder.class);
-
-        dispatcher = (DefaultSecDispatcher) container.lookup(SecDispatcher.class, "maven");
+        return container;
     }
 
-    private List<CoreExtensionEntry> loadCoreExtensions(List<CoreExtension> extensions, ClassRealm containerRealm,
-            Set<String> providedArtifacts) {
+    private static List<CoreExtensionEntry> loadCoreExtensions(
+            List<CoreExtension> extensions,
+            ClassRealm containerRealm,
+            Set<String> providedArtifacts,
+            ClassWorld classWorld,
+            Slf4jLoggerManager plexusLoggerManager,
+            Logger slf4jLogger,
+            ILoggerFactory slf4jLoggerFactory,
+            Function<CliRequest, TransferListener> createTransferListener) {
```

Comment:
What's the benefit of using a static method and passing all parameters ?

Reply:
Actually now, that those fields are final, you are right, the method could be non-static without any danger that we inadvertently change the state. Let me change it.